src/sys/unix.rs

```diff
@@ -195,11 +199,7 @@ pub fn read_stdin(arena: &Arena, mut timeout: time::Duration) -> Option<ArenaStr
                 let beg = time::Instant::now();
 
                 let mut pollfd = libc::pollfd { fd: STATE.stdin, events: libc::POLLIN, revents: 0 };
-                let ts = libc::timespec {
-                    tv_sec: timeout.as_secs() as libc::time_t,
-                    tv_nsec: timeout.subsec_nanos() as libc::c_long,
-                };
-                let ret = libc::ppoll(&mut pollfd, 1, &ts, null());
```

Comment:
I'd like to keep the old `ppoll` code for Linux. I think you could do it with something like
```rs
let ret;

#[cfg(target_os = "macos")]
{
    ret = libc::poll(&mut pollfd, 1, timeout.as_millis() as libc::c_int);
}
#[cfg(not(target_os = "macos"))]
{
    let ts = libc::timespec {
        tv_sec: timeout.as_secs() as libc::time_t,
        tv_nsec: timeout.subsec_nanos() as libc::c_long,
    };
    ret = libc::ppoll(&mut pollfd, 1, &ts, null());
}
```

Not sure if there's a better approach.